spark-benchmarks/src/main/scala/AdvertisingSpark.scala

```diff
@@ -91,20 +91,18 @@ object KafkaRedisAdvertisingStream {
     //each record in the RDD: key:(campaign_id : String, window_time: Long),  Value: (ad_id : String)
     //DStream[((String,Long),String)]
 
-    val groupedByCampaignTime = campaign_timeStamp.groupByKey()
-    // DStream[((String,Long), Iterable[String])]
+    // since we're just counting use reduceByKey
+    val totalEventsPerCampaignTime = campaign_timeStamp.mapValues(_ => 1).reduceByKey(_ + _)
```

Comment:
Yes, this is that I say, I think that is more efficient.

I'm very interested in the test after this PR.

I can see the code more time... if we can reduce the processing time in Spark the distance with Flink and Storm is better for the Spark lovers. :)

Regards.